### PR TITLE
Msp pfb and rvm fitting

### PR DESCRIPTION
**Pull request by keegansmith21**
Large update that handles inverse pfb files and fits the RVM. The RVM fits are still wring by a fixed amount in PA, but there's nothing I can do about this now as it's a psrmodel developer issue, whom I have contacted.

### CODE REVIEW

File: plotting/plotting_toolkit.py [hunk 1]
```diff
@@ -22,29 +25,82 @@
...
-    #from mwa_pulsar_client import client
-    #address = "https://mwa-pawsey-volt01.pawsey.org.au"
-    #auth = ('mwapulsar','veovys9OUTY=')
-    #pul_list_dict = client.pulsar_list(address, auth)
-    #detection_list = client.detection_list(address, auth)
-    #return_stuff = client.detection_get(address, auth, observationid=obsid)
-    #print(client.pulsar_get(address, auth, name="J2241-5236"))
+def read_ascii_archive(archive):
+    """
+    Reads an ascii archive and calculates some stuff
```
Comment (NickSwainston) [inline plotting/plotting_toolkit.py]: some stuff is vague

File: scripts/check_known_pulsars.py [hunk 2]
```diff
@@ -30,6 +32,102 @@
...
     ATNF_LOC = None
 
+def search_for_cal_srclist(obsid, cal_id, all_cal_returns=False, all_srclist_returns=False):
+    """
+    Given an obsid, searches common locations for the rts folder as well as the sourcelist
+
+    Parameters:
+    -----------
+    obsid: int
+        The observation ID
```
Comment (NickSwainston) [inline scripts/check_known_pulsars.py:42]: missing input paramaters

File: scripts/check_known_pulsars.py [hunk 3]
```diff
@@ -30,6 +32,102 @@
...
+                my_srclist = srclists[choice-1]
+                print("Using sourcelist directory: {}".format(my_srclist))
+                srclists = [my_srclist]
+            else:
+                print("## Not a valid choice! ##")
+
+    return cal_dirs, srclists
+
+def upload_cal_files(obsid, cal_id, cal_dir_to_tar=None, srclist=None):
+
```
Comment (NickSwainston) [inline scripts/check_known_pulsars.py:107]: Needs a docstring

File: scripts/check_known_pulsars.py [hunk 4]
```diff
@@ -30,6 +32,102 @@
...
+
+    if not cal_dir_to_tar:
+        cal_dir_to_tar = search_for_cal_srclist(obsid, cal_id, all_srclist_returns=True)[0][0]
+    if not srclist:
+        srclist = search_for_cal_srclist(obsid, cal_id, all_cal_returns=True)[1][0]
```
Comment (NickSwainston) [inline scripts/check_known_pulsars.py]: Should only do this is there isn't already a calibration online (put it in the else at line 124)

**Review by NickSwainston — CHANGES_REQUESTED** [review]
Admittedly I skimmed most of this but I found a few little things to fix up. Overall looks like well-documented code

File: plotting/plotting_toolkit.py (hunk 1 above)
Comment (keegansmith21) [reply plotting/plotting_toolkit.py]: updated

File: scripts/check_known_pulsars.py (hunk 2 above)
Comment (keegansmith21) [reply scripts/check_known_pulsars.py:42]: updated

File: scripts/check_known_pulsars.py (hunk 3 above)
Comment (keegansmith21) [reply scripts/check_known_pulsars.py:107]: updated

File: scripts/check_known_pulsars.py (hunk 4 above)
Comment (keegansmith21) [reply scripts/check_known_pulsars.py]: updated

**Review by NickSwainston — APPROVED** [review]

**Pull request merged**